import "goby/core/proto/option_extensions.proto";
import "goby/core/proto/app_base_config.proto";

message Sub
{

    required double baz = 1;
    required double bar = 2;  
    repeated uint32 rint = 3;
}

message TestMessage
{
    required double foo = 1;
    optional string name = 2;
    enum MyEnum { 
         YES = 1;
         NO = 2;
         MAYBE = 3;
     }
    optional MyEnum evalue = 3;
    optional Sub subfoo = 4;
    repeated string rstr = 5;
}

message TestGConfig
{
  required AppBaseConfig base = 1;
  
  required double dvalue1 = 2 [(description)="some fun double"];
  optional string svalue2 = 3;    
  
  enum MyEnum { 
    YES = 1;
    NO = 2;
    MAYBE = 3;
  }
  required MyEnum evalue3 = 4;
  
  repeated int32 i32 = 5;
  repeated int64 i64 = 6;
  repeated uint32 ui32 = 7;
  repeated uint64 ui64 = 8;
}
 
